app/pulp/app/serializers/progress.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework import serializers
+
+from pulp.app import models
+from pulp.app.serializers import ModelSerializer
+
+
+class ProgressReportSerializer(ModelSerializer):
+
+    _href = serializers.HyperlinkedIdentityField(
+    view_name='progress-detail',
+    lookup_field='id',
+    )
+
+    message = serializers.CharField(
+        help_text="The message shown to the user for the progress update.",
+        read_only=True
+    )
+
+    state = serializers.CharField(
+        help_text="The current state of the progress update. The possible values include:"
+                  " 'waiting', 'skipped', 'running', 'completed', 'failed' and 'canceled'."
+                  " Default is waiting.",
+        read_only=True
+    )
+
+    total = serializers.IntegerField(
+        help_text="The total count of items to be handled by the ProgressBar."
+    )
+
+    done = serializers.IntegerField(
+        help_text="The count of items already processed. Defaults to 0."
+    )
+
+    task = serializers.HyperlinkedRelatedField(
+        help_text="The task associated with this progress report.",
+        read_only=True,
+        view_name='tasks-detail'
+    )
+
+    class Meta:
+        model = models.ProgressReport
+        fields = ModelSerializer.Meta.fields + ('message', 'state', 'total', 'done', 'task')
+
+
+class ProgressSpinnerSerializer(ProgressReportSerializer):
```

Comment:
I don't think we need this serializer. We only need to serialize the ProgressReport model since the other two model types, `ProgressBar`, and `ProgressSpinner` are proxy models of `ProgressReport`.